include/exadg/solvers_and_preconditioners/multigrid/transfers/mg_transfer_c.cpp

```diff
@@ -96,20 +96,30 @@ MGTransferC<dim, Number, VectorType, components>::do_restrict_and_add(VectorType
   FEEvaluation<dim, degree, 1, components, Number> fe_eval_cg(data_composite, 0);
   FEEvaluation<dim, degree, 1, components, Number> fe_eval_dg(data_composite, 1);
 
+  VectorType src_temp;
+  data_composite.initialize_dof_vector(src_temp, 1);
+  src_temp.copy_locally_owned_data_from(src);
```

Comment:
Do we need this one? In case we never use anything but an actual DG space here, all DoFs on locally owned cells will be locally owned, so we do not need the copy here. The real difference should be in the CG space where the ghost space might or might not match. (The same is true for the prolongate function below.) One side effect would be that the more expensive vector with more DoFs does not get copied here, saving memory transfer.

Reply:
OK. I was able to remove a set of temporal vectors.

Reply:
Now I'm confused, because I would have expected the exact opposite to be possible, i.e., use no temporary vector for `src` in `restrict` (which corresponds to the DG space), but a temporary vector for `dst` (the continuous FEM space), whereas I would expect a temporary vector for `src` in `prolongate`. Can you check whether the current version really works with MPI?

Reply:
Yes, this is working. The problem is that the DG vector has ghost values, but MatrixFree/FEEvaluation is expecting a vector without ghost values. I guess this is related to the `AdditionalData`. Playing around with it could be an option, but also a bit brittle, since ghosting also depends on whether all faces are owned...

Reply:
I see, thanks for the background. So it is rather the assertion in https://github.com/dealii/dealii/blob/e37add0c8852ca92525961c144fc2df54a146029/include/deal.II/matrix_free/fe_evaluation.h#L3377-L3393 that is too conservative: If all indices on the present cell are indeed locally owned (not ghosted), we would only need to check whether the locally owned range coincides, as the locally owned part is guaranteed to be ok. While trying to get this more precise in deal.II would be useful to some context, I guess we should switch this to the deal.II global coarsening setup at some point anyway, so we can keep this for now and switch later.